python/ray/tune/integration/pytorch_lightning.py

```diff
@@ -214,10 +220,14 @@ def __init__(self,
         super(_TuneCheckpointCallback, self).__init__(on)
         self._filename = filename
 
-    def _handle(self, trainer: Trainer, pl_module: LightningModule):
+    def _handle(self,
+                trainer: Trainer,
+                pl_module: LightningModule,
+                is_epoch: bool = False):
         if trainer.running_sanity_check:
             return
-        with tune.checkpoint_dir(step=trainer.global_step) as checkpoint_dir:
+        step = trainer.current_epoch if is_epoch else trainer.global_step
+        with tune.checkpoint_dir(step=step) as checkpoint_dir:
```

Comment:
will it be possible that you can have `global_step` and `current_epoch` override each other in the same training loop?

Reply:
Hmm I don’t think that’s possible as long as you’re using 1 callback since it’ll checkpoint at either the epoch frequency or batch frequency.

Reply:
But maybe I’m misunderstanding what you are asking

Reply:
ok cool

Reply:
Hm, actually we do allow passing multiple `on` values to a single callback. So in that case it might interfere. 
I'm not sure this is a common use case (mixing on-step and on-epoch checkpointing) but we might want to ensure we don't run into conflicts.

Things we could do: First, use a formatted string instead of an int to pass to `tune.checkpoint_dir(step)` (which is actually supported). Second, set `self._epoch = True` if any of the `on` indicate per-epoch checkpointing and always use epochs for checkpointing (potentially overwriting existing checkpoints at the end of an epoch). Third, disallow multiple `on` values for checkpoints

Reply:
Fourth, like thirds but throw a warning instead. I'd be fine with that.

Reply:
Right, passing in multiple values to `on` would be pretty uncommon but is still possible. I'll change it to option 1- the default PTL callbacks do the same thing.

Reply:
I made the change @krfricke- please take another look.